### DISCUSSION TITLE
How to export small strain component using auxkernel RankTwoAux?

### DISCUSSION THREAD

**Discussion by ZhigangPu** [idaholab/moose#19101] [Q&A Modules: Solid mechanics] [answered]
Dear,

I want to export strain tensor component using `RankTwoAux` auxkernel. How can I do that? I only know the way with stress like below:
Aux variable definition:
![image](https://user-images.githubusercontent.com/24690469/137266737-5de06654-3aaa-44e9-8a50-66a0cbf3b86d.png)
Aux kernel :
![image](https://user-images.githubusercontent.com/24690469/137266757-6f48abd0-fa65-42ed-ab05-2ec88fc0ae4e.png)
Material stress tensor calculation:
![image](https://user-images.githubusercontent.com/24690469/137266782-0dfbc9fb-2d2b-4999-8b0e-f0ecfacaa4a1.png)
I suppose the logic is first define aux variable, then use auxkernels `RankTwoAux` to get tensor component for the variable, while the tensor is come from the material object `ComputeLinearElasticStress`. But I don't know how to use the same logic in the strain.

Any response will be highly appreciated!
zhigang

**Comment by GiudGiud** [comment, marked as answer]
Hello

You should have a look at the tests in the tensor mechanics module. 
Like these ones:
https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/ad_anisotropic_plasticity/anis_plasticity_test.i
https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/global_strain/global_strain_disp.i

@jiangwen84 will probably know a better one for small strain.

Guillaume

**Reply by jiangwen84** [reply]
@GiudGiud  pointed you to the right examples. 

Simplest way is to use TensorMechanicsAction

[Modules/TensorMechanics/Master]
  [all]
    strain = SMALL
    add_variables = true
    incremental = true
    generate_output = 'elastic_strain_xx elastic_strain_yy elastic_strain_xy stress_xx stress_xy stress_yy'
  []
[]

**Reply by ZhigangPu** [reply]
Sorry for late reply! Too many examinations these days.
Many thanks for your help, I will try that!

Best,
zhigang
